packages/lesswrong/lib/collections/comments/schema.ts

```diff
@@ -110,6 +110,20 @@ const schema: SchemaType<"Comments"> = {
     canCreate: ['members'],
     hidden: true,
   },
+  // If this comment is associated with a forum event, the _id of the forumEvent.
+  forumEventId: {
+    ...foreignKeyField({
+      idFieldName: "forumEventId",
+      resolverName: "forumEvent",
+      collectionName: "ForumEvents",
+      type: "ForumEvent",
+      nullable: true,
+    }),
+    optional: true,
+    canRead: ['guests'],
+    canCreate: ['members'],
+    hidden: true,
+  },
```

Comment:
I tried putting this in the `publicData` along with the vote position, which would have avoided this schema change. I found that it was hard to avoid bugs with linking/unlinking or refetching that way, because the vote position and comment can be changed independently which creates a lot of cases